chainer/functions/array/stack.py

```diff
@@ -1,4 +1,36 @@
-import chainer
+from chainer import cuda
+from chainer import function_node
+from chainer.functions.array import separate
+from chainer.utils import type_check
+
+
+class Stack(function_node.FunctionNode):
+
+    """Concatenate variables along a new axis."""
+
+    def __init__(self, axis):
+        self.axis = axis
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() > 0)
+        type_check.expect(
+            -in_types[0].ndim - 1 <= self.axis,
+            self.axis <= in_types[0].ndim
+        )
+        dtype = in_types[0].dtype
+        shape = in_types[0].shape
+        for x_type in in_types[1:]:
+            type_check.expect(
+                x_type.dtype == dtype,
+                x_type.shape == shape,
+            )
+
+    def forward(self, inputs):
+        xp = cuda.get_array_module(*inputs)
+        return xp.stack(inputs, axis=self.axis),
```

Comment:
numpy.stack is not supported in older version of NumPy.
Please avoid it, by using `numpy.concatenate`, for example.